client-hc/src/main/java/com/graphhopper/api/GraphHopperWeb.java

```diff
@@ -341,9 +354,9 @@ private Request createGetRequest(GHRequest ghRequest) {
             url += "&key=" + WebHelper.encodeURL(key);
         }
 
-        for (Map.Entry<String, String> entry : ghRequest.getHints().toMap().entrySet()) {
+        for (Map.Entry<String, Object> entry : ghRequest.getHints().toMap().entrySet()) {
             String urlKey = entry.getKey();
-            String urlValue = entry.getValue();
+            String urlValue = entry.getValue().toString();
```

Comment:
Here I also wonder in which cases the `toString()` method really produces something useful. (In this case it must be valid to use as url parameter).

Reply:
As we use WebHelper.encodeURL this should be fine (?)

Reply:
Hm, yeah encoding the url probably replaces invalid characters, but the real question is will this ever be useful on the server side? Sure `toString()` *might* produce something that can be parsed later, but that entirely depends on the object we put in (which seems a bit fragile and forces one to implement `toString()` in a certain way (which I do not like so much)). Then again I do not have a much better idea :) But maybe we can throw an error if the values is not a number and not a string (are there any actual use cases where we want to `toString()` an object into a url parameter?)

Reply:
As we use by default a POST request this should not make a problem IMO. Throwing an error is a bit clumsy as we need to check for boolean, short, byte, int, double, ...

Reply:
Hm, but this whole method is about creating the url parameters for a GET request. If we do not use it and only use POST we should rather remove the entire method. Checking for 'number, boolean, string or something else' might take a few lines to do but that should not be a problem. Then again if you think using `toString()` is fine we can just as well keep it :) (I just do not really know in which case it would be useful and do not want to encourage people to adjust their `toString()` implementation to make it work with this url creation code).